mesheryctl/internal/cli/root/components/view.go

```diff
@@ -0,0 +1,170 @@
+	Short: "view registered components",
+	Long:  "view a component registered in Meshery Server",
+	Example: `
+// View details of a specific component
+mesheryctl exp components view [component-name]
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		//Check prerequisite
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return err
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	Args: func(_ *cobra.Command, args []string) error {
+		const errMsg = "Usage: mesheryctl exp component view [component-name]\nRun 'mesheryctl exp component view --help' to see detailed help message"
+		if len(args) == 0 {
+			return fmt.Errorf("component name isn't specified\n\n%v", errMsg)
+		} else if len(args) > 1 {
+			return fmt.Errorf("too many arguments\n\n%v", errMsg)
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			log.Fatalln(err, "error processing config")
+		}
+
+		baseUrl := mctlCfg.GetBaseMesheryURL()
+		component := args[0]
+
+		url := fmt.Sprintf("%s/api/meshmodels/components?search=%s&pagesize=all", baseUrl, component)
+		req, err := utils.NewRequest(http.MethodGet, url, nil)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		resp, err := utils.MakeRequest(req)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		// defers the closing of the response body after its use, ensuring that the resources are properly released.
+		defer resp.Body.Close()
+
+		data, err := io.ReadAll(resp.Body)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		componentResponse := &models.MeshmodelComponentsAPIResponse{}
+		err = json.Unmarshal(data, componentResponse)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		var selectedComponent v1alpha1.ComponentDefinition
+
+		if componentResponse.Count == 0 {
+			fmt.Println("No component(s) found for the given name: ", component)
+			return nil
+		} else if componentResponse.Count == 1 {
+			selectedComponent = componentResponse.Components[0] // Update the type of selectedModel
+		} else {
+			selectedComponent = selectComponentPrompt(componentResponse.Components)
+		}
+
+		var output []byte
+
+		// user may pass flag in lower or upper case but we have to keep it lower
+		// in order to make it consistent while checking output format
+		outFormatFlag = strings.ToLower(outFormatFlag)
+		if outFormatFlag == "yaml" {
+			if output, err = yaml.Marshal(selectedComponent); err != nil {
+				return errors.Wrap(err, "failed to format output in YAML")
+			}
+			if saveFlag {
+				fmt.Println("Saving output as YAML file")
+				err = os.WriteFile("./output.yaml", output, 0666)
```

Comment:
Give the file name same as component's name otherwise file will be overwritten. My suggestion is change the location to .meshery/content folder.

Reply:
Valid point, I'll change this up 👍